### PR TITLE
PMM-3516 Dashboard provisioning

### PR DESCRIPTION
**Pull request by adivinho**

### CODE REVIEW

File: import-dashboards.py [hunk 1]
```diff
@@ -19,16 +19,18 @@
...
+SCRIPT_DIR            = os.path.dirname(os.path.abspath(__file__))
+DASHBOARD_DIR         = SCRIPT_DIR + '/dashboards/'
+NEW_VERSION_FILE      = SCRIPT_DIR + '/VERSION'
+OLD_VERSION_FILE      = GRAFANA_DB_DIR + '/PERCONA_DASHBOARDS_VERSION'
+GRAFANA_PROVISION_DIR = '/usr/share/grafana/conf/provisioning/dashboards/'
+PMM_PLUGIN_DIR        = '/var/lib/grafana/plugins/pmm-app/dist/dashboards/'
+HOST                  = 'http://127.0.0.1:3000'
+LOGO_FILE             = '/usr/share/pmm-server/landing-page/img/pmm-logo.svg'
+SET_OF_TAGS 	      = {'QAN': 0, 'OS': 0, 'MySQL': 0, 'MongoDB': 0, 'PostgreSQL': 0, 'HA': 0, 'Cloud': 0, 'Insight': 0, 'PMM': 0}
+CONTENT               = '''<center>
```
Comment (askomorokhov) [inline import-dashboards.py]: Such align is dis-recommended by [PEP 8/Pet Peeves](https://www.python.org/dev/peps/pep-0008/#pet-peevesl)

> Avoid extraneous whitespace in the following situations:
>   More than one space around an assignment (or other) operator to align it with another.

File: import-dashboards.py [hunk 2]
```diff
@@ -242,6 +244,47 @@ def copy_apps():
...
+        except OSError:
+            print ' * Creation of subfolder %s failed' % path
+        else:
+            print ' * Successfully created subfolder %s' % path
+
+
+def move_dashboards_into_folders():
+    for filename in os.listdir(PMM_PLUGIN_DIR):
+        if os.path.isfile(os.path.join(PMM_PLUGIN_DIR, filename)):
+            with open(PMM_PLUGIN_DIR + filename, 'r') as dashboard_file:
```
Comment (askomorokhov) [inline import-dashboards.py]: I think it's better to do `os.path.join(PMM_PLUGIN_DIR, filename)` ones abowbe `if`

File: import-dashboards.py [hunk 3]
```diff
@@ -242,6 +244,47 @@ def copy_apps():
...
+def create_provisioning_configs():
+    for folder in SET_OF_TAGS.keys():
+        try:
+            f = open(GRAFANA_PROVISION_DIR + folder + '.yaml',"w+")
+            f.write('apiVersion: 1\n\nproviders:\n - name: %s\n   orgId: 1\n   folder: %s\n   type: file\n   updateIntervalSeconds: 360\n' % (folder,folder))
+            f.write('   options:\n     path: %s\n' % ('/var/lib/grafana/plugins/pmm-app/dist/dashboards/' + folder))
+            f.close()
+            print ' * Created provisioning config file %s.yaml' % (GRAFANA_PROVISION_DIR + folder)
+        except:
+            continue
```
Comment (askomorokhov) [inline import-dashboards.py]: `continue` looks hear rudimental. What do you think to print what error happened? Could be useful for debug.

File: import-dashboards.py [hunk 4]
```diff
@@ -242,6 +244,47 @@ def copy_apps():
...
+                if tag == 'Percona':
+                    try:
+                        tag = dashboard['tags'][1]
+                    except:
+                        continue
+                try:
+                    shutil.move(os.path.join(PMM_PLUGIN_DIR, filename), PMM_PLUGIN_DIR + tag)
+                    print ' * Moved dashboard %s into subfolder %s' % (dashboard['title'], PMM_PLUGIN_DIR + tag)
+                except:
+                    continue
```
Comment (askomorokhov) [inline import-dashboards.py]: same comment as below:`continue` -> `print`

File: import-dashboards.py (hunk 1 above)
Comment (adivinho) [reply import-dashboards.py]: Fixed

File: import-dashboards.py (hunk 3 above)
Comment (adivinho) [reply import-dashboards.py]: Added print out of an exception

File: import-dashboards.py (hunk 4 above)
Comment (adivinho) [reply import-dashboards.py]: Fixed

File: import-dashboards.py (hunk 2 above)
Comment (adivinho) [reply import-dashboards.py]: Added

**Comment by AlekSi** [comment]
What should we do there?

**Comment by rnovikovP** [comment]
Let's merge this when no conflicts. this is one of the steps we need to remove API keys

**Comment by atymchuk** [comment]
Closing it as no longer relevant. Dashboards are provisioned with the same mechanism.

**Pull request closed without merging**